### PR TITLE
Timeout client

### PR DESCRIPTION
**Pull request by zhujun98**
Add timeout to `Client`. This MR is based on #20 and #23.

### CODE REVIEW

File: include/karabo-bridge/kb_client.hpp [hunk 1]
```diff
@@ -629,11 +629,33 @@ class Client {
...
+    Client(const Client&) = delete;
+    Client& operator=(const Client&) = delete;
 
     void connect(const std::string& endpoint) {
         std::cout << "Connecting to server: " << endpoint << std::endl;
-        socket_.connect(endpoint.c_str());
+        socket_.connect(endpoint);
+    }
+
+    void close() {
```
Comment (zhujun98) [inline include/karabo-bridge/kb_client.hpp]: I am sure whether we need this one. I am using it to shutdown a client with infinite timeout in the unittest. It causes SIGABRT roughly 1 out 10 runs. But I think creating and using `socket` in different threads is not encouraged. I need to brush up my knowledge about zmq.

`pfd.revents & POLLIN (src/signaler.cpp:264)`

**Comment by zhujun98** [comment]
> I've only reviewed the C++ part, which LGTM, but I can't tell with regards to the docs

Thanks! See you next month :-)

File: include/karabo-bridge/kb_client.hpp [hunk 2]
```diff
@@ -629,11 +640,28 @@ class Client {
     }
 
 public:
-    Client(): ctx_(1), socket_(ctx_, ZMQ_REQ) {}
+    /*
+     * Constructor.
+     *
+     * @param timeout: connection timeout in milliseconds. "-1" (default) for infinite.
+     */
+    explicit Client(int timeout=-1): ctx_(1), socket_(ctx_, ZMQ_REQ) {
```
Comment (tmichela) [inline include/karabo-bridge/kb_client.hpp]: Minor suggestion: the timeout parameter unit is `second` in the python client. We should maybe keep it the same here to stay consistent.

Comment (zhujun98) [reply include/karabo-bridge/kb_client.hpp]: Changed. But maybe we should have kept the default value in milliseconds at the beginning.

**Comment by zhujun98** [comment]
Thanks for the review! If there is no further comment, I will merge it later today.

File: include/karabo-bridge/kb_client.hpp (hunk 2 above)
Comment (tmichela) [reply include/karabo-bridge/kb_client.hpp]: I originally had it in milliseconds, Thomas and Hans argued for using seconds instead:
https://github.com/European-XFEL/karabo-bridge-py/pull/46#discussion_r239425916

Comment (zhujun98) [reply include/karabo-bridge/kb_client.hpp]: Thanks for the explanation! It's life.

**Pull request merged**